pkg/deploy/predeploy_test.go

```diff
@@ -0,0 +1,1949 @@
+			},
+		},
+	}
+
+	type mock func(*mock_features.MockDeploymentsClient, *mock_features.MockResourceGroupsClient, *mock_msi.MockUserAssignedIdentitiesClient, *mock_keyvault.MockManager, *mock_compute.MockVirtualMachineScaleSetsClient, *mock_compute.MockVirtualMachineScaleSetVMsClient)
+	genericSubscriptionDeploymentFailed := func(d *mock_features.MockDeploymentsClient, rg *mock_features.MockResourceGroupsClient, m *mock_msi.MockUserAssignedIdentitiesClient, k *mock_keyvault.MockManager, vmss *mock_compute.MockVirtualMachineScaleSetsClient, vmssvms *mock_compute.MockVirtualMachineScaleSetVMsClient) {
+		d.EXPECT().CreateOrUpdateAtSubscriptionScopeAndWait(ctx, gomock.Any(), gomock.Any()).Return(
+			errors.New("generic error"),
+		).AnyTimes()
+	}
+	subscriptionDeploymentSuccessful := func(d *mock_features.MockDeploymentsClient, rg *mock_features.MockResourceGroupsClient, m *mock_msi.MockUserAssignedIdentitiesClient, k *mock_keyvault.MockManager, vmss *mock_compute.MockVirtualMachineScaleSetsClient, vmssvms *mock_compute.MockVirtualMachineScaleSetVMsClient) {
+		d.EXPECT().CreateOrUpdateAtSubscriptionScopeAndWait(ctx, gomock.Any(), gomock.Any()).Return(nil)
+	}
+	subscriptionRGDeploymentFailed := func(d *mock_features.MockDeploymentsClient, rg *mock_features.MockResourceGroupsClient, m *mock_msi.MockUserAssignedIdentitiesClient, k *mock_keyvault.MockManager, vmss *mock_compute.MockVirtualMachineScaleSetsClient, vmssvms *mock_compute.MockVirtualMachineScaleSetVMsClient) {
+		d.EXPECT().CreateOrUpdateAndWait(ctx, subscriptionRgName, gomock.Any(), gomock.Any()).Return(
+			errors.New("generic error"),
+		)
+	}
+	globalRGDeploymentFailed := func(d *mock_features.MockDeploymentsClient, rg *mock_features.MockResourceGroupsClient, m *mock_msi.MockUserAssignedIdentitiesClient, k *mock_keyvault.MockManager, vmss *mock_compute.MockVirtualMachineScaleSetsClient, vmssvms *mock_compute.MockVirtualMachineScaleSetVMsClient) {
+		d.EXPECT().CreateOrUpdateAndWait(ctx, globalRgName, gomock.Any(), gomock.Any()).Return(
+			errors.New("generic error"),
+		)
+	}
+	gatewayRGDeploymentFailed := func(d *mock_features.MockDeploymentsClient, rg *mock_features.MockResourceGroupsClient, m *mock_msi.MockUserAssignedIdentitiesClient, k *mock_keyvault.MockManager, vmss *mock_compute.MockVirtualMachineScaleSetsClient, vmssvms *mock_compute.MockVirtualMachineScaleSetVMsClient) {
+		d.EXPECT().CreateOrUpdateAndWait(ctx, gatewayRgName, gomock.Any(), gomock.Any()).Return(
+			errors.New("generic error"),
+		)
+	}
+	rpRGDeploymentFailed := func(d *mock_features.MockDeploymentsClient, rg *mock_features.MockResourceGroupsClient, m *mock_msi.MockUserAssignedIdentitiesClient, k *mock_keyvault.MockManager, vmss *mock_compute.MockVirtualMachineScaleSetsClient, vmssvms *mock_compute.MockVirtualMachineScaleSetVMsClient) {
+		d.EXPECT().CreateOrUpdateAndWait(ctx, rpRgName, gomock.Any(), gomock.Any()).Return(
+			errors.New("generic error"),
+		)
+	}
```

Comment:
This is not how you should do that, in your tests you should have the data that you feed your functions instead of this.

Reply:
Updated the test cases with test data getting passed from the test cases.